projects/packages/forms/src/contact-form/class-contact-form-field.php

```diff
@@ -2044,4 +2056,116 @@ private function has_inset_label() {
+	 * selected rating in a hidden input so it is handled by existing form
+	 * validation/submission logic.
+	 *
+	 * @since 0.46.0
+	 *
+	 * @param string $id                 Field ID.
+	 * @param string $label              Field label.
+	 * @param string $value              Current value.
+	 * @param string $class              Additional CSS classes.
+	 * @param bool   $required           Whether field is required.
+	 * @param string $required_field_text Required label text.
+	 * @return string HTML markup.
+	 */
+	private function render_rating_field( $id, $label, $value, $class, $required, $required_field_text ) {
+		// Enqueue stylesheet for rating field.
+		wp_enqueue_style( 'jetpack-form-field-rating-style', plugins_url( '../../dist/blocks/field-rating/style.css', __FILE__ ), array(), Constants::get_constant( 'JETPACK__VERSION' ) );
+
+		// Read block attributes needed for rendering.
+
+		$max_attr = $this->get_attribute( 'max' );
+
+		$max_rating = is_numeric( $max_attr ) && (int) $max_attr > 0 ? (int) $max_attr : 5;
+
+		$initial_rating = (int) $value ? (int) $value : 0;
+
+		$label_html = $this->render_label( 'rating', $id, $label, $required, $required_field_text );
+
+		$spans = '';
+		for ( $i = 1; $i <= $max_rating; $i++ ) {
+			$spans .= sprintf(
+				'<label class="jetpack-field-rating__label">
+					<input
+						class="jetpack-field-rating__input"
+						type="radio"
+						data-wp-on--change="actions.onFieldChange"
+						%1$s
+						%2$s
+						name="%3$s"
+						value="%4$s/%5$s" />
+				</label>',
+				checked( $i, $initial_rating, false ),
+				$required ? 'required aria-required="true"' : '',
+				esc_attr( $id ),
+				esc_attr( $i ),
+				esc_attr( $max_rating )
+			);
+		}
+
+		$style_attr = '';
+
+		$css_styles = array_filter( array_map( 'trim', explode( ';', $this->field_styles ) ) );
+
+		$css_key_value_pairs = array_reduce(
+			$css_styles,
+			function ( $pairs, $style ) {
+				list( $key, $value )   = explode( ':', $style );
+				$pairs[ trim( $key ) ] = trim( $value );
+				return $pairs;
+			},
+			array()
+		);
+
+		// The rating input overwrites the text color, so we are using a custom logic to set the star color as a CSS variable.
+		$has_star_color = isset( $css_key_value_pairs['color'] );
+
+		if ( $has_star_color ) {
+			$color_value = $css_key_value_pairs['color'];
+			$style_attr  = 'style="--jetpack--contact-form--rating-star-color: ' . esc_attr( $color_value ) . ';';
+			unset( $css_key_value_pairs['color'] );
+		} else {
+			// Theme colors are set in the field_classes attribute
+			$preset_colors = array(
+				'has-base-color'     => '--wp--preset--color--base',
+				'has-contrast-color' => '--wp--preset--color--contrast',
+			);
+
+			if ( preg_match( '/has-accent-(\d+)-color/', $this->field_classes, $matches ) ) {
+				$accent_number = $matches[1];
+				$preset_colors[ 'has-accent-' . $accent_number . '-color' ] = '--wp--preset--color--accent-' . $accent_number;
+			}
```

Comment:
There's a lot of complex css logic here, and it seems like it could be brittle. I'm having a hard time following it, and it may be needed, but just seems un-ideal.